### PR TITLE
doc: improve the documentation landing page 

### PR DESCRIPTION
**Pull request by annastuchlik**
This PR introduces the following changes to the documentation landing page:

- The " New to ScyllaDB? Start here!" box is added.
- The "Connect your application to Scylla" box is removed.
- Some wording has been improved.
- "Scylla" has been replaced with "ScyllaDB".

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -57,75 +59,75 @@
 
   <div class="topics-grid topics-grid--products">
 
-      <h2 class="topics-grid__title">Our Product List</h2>
-      <p class="topics-grid__text">To begin choose a product from the list below</p>
+      <h2 class="topics-grid__title">Our Products</h2>
+      <p class="topics-grid__text">Choose a product from the list below to view its documentation.</p>
```
Comment (tzach) [inline docs/index.rst:63]: This sentence brings zero additional info
Let's just remove it 
```suggestion
```

**Comment by tzach** [comment]
LGTM

**Comment by annastuchlik** [comment]
@scylladb/scylla-maint Please backport to 5.1

**Comment by denesb** [comment]
> @scylladb/scylla-maint Please backport to 5.1

Done.

**Pull request merged**